src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -27,35 +28,84 @@
     private static final int MAX_BUFFER_SIZE = 1024 * 8;
 
     @Function
-    public static byte[] routeEx(
-        String address,
-        int port)
+    public static SocksRouteExBuilder routeEx()
     {
-        final MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
-        final SocksRouteExFW routeEx = new SocksRouteExFW.Builder()
-                                                         .wrap(writeBuffer, 0, writeBuffer.capacity())
-                                                         .address(address)
-                                                         .port(port)
-                                                         .build();
-        final byte[] bytes = new byte[routeEx.sizeof()];
-        routeEx.buffer().getBytes(0, bytes);
-        return bytes;
+        return new SocksRouteExBuilder();
     }
 
     @Function
-    public static byte[] beginEx(
-        String address,
-        int port)
+    public static SocksBeginExBuilder beginEx()
+    {
+        return new SocksBeginExBuilder();
+    }
+
+
+    public static final class SocksRouteExBuilder
     {
-        final MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
-        final SocksBeginExFW beginEx = new SocksBeginExFW.Builder()
-                                                         .wrap(writeBuffer, 0, writeBuffer.capacity())
-                                                         .address(address)
-                                                         .port(port)
-                                                         .build();
-        final byte[] bytes = new byte[beginEx.sizeof()];
-        beginEx.buffer().getBytes(0, bytes);
-        return bytes;
+        private final SocksRouteExFW.Builder routeExRw;
+
+        private SocksRouteExBuilder()
+        {
+            MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
+            this.routeExRw = new SocksRouteExFW.Builder().wrap(writeBuffer, 0, writeBuffer.capacity());
+        }
+
+        public SocksRouteExBuilder address(String address)
```

Comment:
Method parameters should be on the next line, please update all methods as necessary.